conan/internal/cache/integrity_check.py

```diff
@@ -17,17 +18,39 @@ class IntegrityChecker:
         This is to be done over the package contents, not the compressed conan_package.tgz
         artifacts
     """
-    def __init__(self, cache):
+    def __init__(self, cache, home_folder):
         self._cache = cache
+        self._pkg_signatures_plugin = PkgSignaturesPlugin(cache, home_folder)
```

Comment:
Does this check means that when the integrity check is called from ``conan upload`` it will try to verify the package signatures before the upload? Is this an intended use case or not?
What should a plugin do if the package that is uploading is already signed? Skip the signature, add its own?